src/components/People/PeopleTable.tsx

```diff
@@ -0,0 +1,190 @@
+import { useSearchParams } from 'react-router-dom';
+import classNames from 'classnames';
+
+import { Person } from '../../types';
+import { PersonRow } from '../Person/PersonRow';
+import { SearchLink } from '../SearchLink';
+import { SearchParams } from '../../utils/searchHelper';
+
+type PeopleTableProps = {
+  people: Person[]
+  clickedPersonSlug?: string
+};
+
+export const PeopleTable = (
+  { people, clickedPersonSlug }: PeopleTableProps,
+) => {
+  const [searchParams] = useSearchParams();
+
+  const query = searchParams.get('query')?.toLowerCase();
+  const centuries = searchParams.getAll('centuries');
+  const sex = searchParams.get('sex');
+  const sort = searchParams.get('sort');
+  const order = searchParams.get('order');
+  let displayedPeople = [...people];
+
+  if (query && query !== null) {
+    displayedPeople = displayedPeople.filter(person => {
+      const personName = person.name.toLowerCase();
+      const motherName = person.motherName?.toLowerCase();
+      const fatherName = person.fatherName?.toLowerCase();
+      const personMatch = personName.includes(query);
+      const motherMatch = motherName?.includes(query);
+      const fatherMatch = fatherName?.includes(query);
+
+      return personMatch || motherMatch || fatherMatch;
+    });
+  }
+
+  if (centuries.length > 0) {
+    displayedPeople = displayedPeople.filter(person => centuries.includes(
+      (Math.floor(person.born / 100) + 1).toString(),
+    ));
+  }
+
+  if (sex !== null) {
+    displayedPeople = displayedPeople.filter(person => person.sex === sex);
+  }
+
+  switch (sort) {
+    case 'name':
+    case 'sex':
+      displayedPeople.sort((p1, p2) => {
+        return order === 'desc'
+          ? p2[sort].localeCompare(p1[sort])
+          : p1[sort].localeCompare(p2[sort]);
+      });
+      break;
+    case 'died':
+    case 'born':
+      displayedPeople.sort((p1, p2) => {
+        return order === 'desc' ? p2[sort] - p1[sort] : p1[sort] - p2[sort];
+      });
+      break;
+    default:
+      break;
+  }
```

Comment:
```suggestion
displayedPeople.sort((p1, p2) => {
  let [person1, person2] = [p1, p2];

  if (order === 'desc') {
    [person2, person1] = [person1, person2];
  }

  switch (sort) {
    case 'name':
    case 'sex':
      return person1[sort].localeCompare(person2[sort]);

    case 'died':
    case 'born':
      return person1[sort] - person2[sort];

    default:
      return 0;
  }
});
```

Reply:
have a look at this implementation.
your option is not a mistake but this is more elegant in my opinion

Reply:
I changed to version with people swap.